compute_tools/src/bin/compute_ctl.rs

```diff
@@ -401,9 +401,8 @@ fn start_postgres(
     }: WaitSpecResult,
 ) -> Result<(Option<PostgresHandle>, StartPostgresResult)> {
     // We got all we need, update the state.
-    let mut state = compute.state.lock().unwrap();
-    state.status = ComputeStatus::Init;
-    compute.state_changed.notify_all();
+    compute.set_status(ComputeStatus::Init);
+    let state = compute.state.lock().unwrap();
```

Comment:
If you swap these lines, you get a deadlock. I'm definitely interested in hearing about any patterns people use to avoid deadlocks in a situation like this. Could I use an unsafe block in a function like `Compute::set_status_unlocked()`?